qa/L0_trace/opentelemetry_unittest.py

```diff
@@ -28,78 +28,112 @@
 
 sys.path.append("../common")
 import json
+import queue
 import re
+import shutil
+import subprocess
+import time
 import unittest
+from functools import partial
 
 import numpy as np
+import requests
 import test_util as tu
 import tritonclient.grpc as grpcclient
 import tritonclient.http as httpclient
+from tritonclient.utils import InferenceServerException
 
-EXPECTED_NUM_SPANS = 16
-# OpenTelemetry OStream exporter sets `parent_span_id` to "0000000000000000",
-# if current span is a root span, i.e. there is no parent span.
-# https://github.com/open-telemetry/opentelemetry-cpp/blob/b7fd057185c4ed2dff507b859cbe058b7609fb4a/exporters/ostream/src/span_exporter.cc#L78C54-L78C68
-NO_PARENT_SPAN = "0000000000000000"
+NO_PARENT_SPAN_ID = ""
+COLLECTOR_TIMEOUT = 10
+
+
+def callback(user_data, result, error):
+    if error:
+        user_data.put(error)
+    else:
+        user_data.put(result)
+
+
+def prepare_data(client, is_binary=True):
+    inputs = []
+    input_data = np.array(
+        [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15], dtype=np.int32
+    )
+    inputs.append(client.InferInput("INPUT0", [1, 16], "INT32"))
+    inputs.append(client.InferInput("INPUT1", [1, 16], "INT32"))
+
+    # Initialize the data
+    input_data = np.expand_dims(input_data, axis=0)
+
+    if is_binary:
+        inputs[0].set_data_from_numpy(input_data)
+        inputs[1].set_data_from_numpy(input_data)
+    else:
+        inputs[0].set_data_from_numpy(input_data, binary_data=is_binary)
+        inputs[1].set_data_from_numpy(input_data, binary_data=is_binary)
```

Comment:
this is the same logically right?
```suggestion
    inputs[0].set_data_from_numpy(input_data, binary_data=is_binary)
    inputs[1].set_data_from_numpy(input_data, binary_data=is_binary)
```

Reply:
suggested code will not work for grpc (tried myself), it mentions that for grpc client `binary_data` is not an option

Reply:
Ah right, I forgot grpcclient was asymmetric like that... No worries then.